### PR TITLE
Fixed: printinfo's 'histogram' output

### PR DESCRIPTION
**Pull request by Shivakumar-SG**
fixes issue #2217.
Let me know if there should be any changes.

### CODE REVIEW

**Comment by Manno15** [comment]
@Shivakumar-SG It looks like the build failed due to not being formatted correctly. You can run something like `mvn package -DskipTests` at the root of the project and it will apply the proper formatting for you. Include those changes into a new commit for this PR and the build should pass from there.

**Comment by Shivakumar-SG** [comment]
I have been trying from past few hours to run `mvn package -DskipTests`, but its throwing me many errors and I couldn't fix them. What would you want me to do?

**Comment by DomGarguilo** [comment]
@Shivakumar-SG what errors are being thrown?

**Comment by DomGarguilo** [comment]
@Shivakumar-SG what errors are being thrown?

**Comment by Shivakumar-SG** [comment]
A whole lot many actually :(

**Comment by Shivakumar-SG** [comment]
if the build failed because of just the formatting, it would be great if someone could help me with the formatted code of my version

**Comment by DomGarguilo** [comment]
> if the build failed because of just the formatting, it would be great if someone could help me with the formatted code of my version

Sure, I can make a PR to your branch with the formatting changes.

**Comment by Shivakumar-SG** [comment]
@DomGarguilo That would be great! Thank you so much

**Comment by EdColeman** [comment]
Can you provide more information?

- What platform are you running on?
- What command are you running?
- What is the first or  first few errors that you are getting?

You can do the same build as the verification checks, but skipping all of the tests by running:

`mvn package javadoc:jar -Psec-bugs -DskipITs=true -DskipTests=true`

The javadoc and sec-bugs will add to the build time, but running those before submitting the PR can highlight issues that will fail the automatic build.  This should also format the files to the community standards.

More information on building is at (Building Accumulo](https://accumulo.apache.org/contributor/building)

**Comment by Manno15** [comment]
It looks like you are missing the import for 'bigNumberForSize()' which is NumUtil.

**Comment by EdColeman** [comment]
It also looks like an extra file (.DS_Store) was pushed along with you commit - that file should not be included.

File: core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java [hunk 1]
```diff
@@ -106,10 +106,11 @@ public void add(int size) {
     }
 
     public void print(String indent) {
-      System.out.println(indent + "Up to size      count      %-age");
+      System.out.println(indent + "Up to size      Count      %-age");
       for (int i = 1; i < countBuckets.length; i++) {
-        System.out.println(String.format("%s%11.0f : %10d %6.2f%%", indent, Math.pow(10, i),
-            countBuckets[i], sizeBuckets[i] * 100. / totalSize));
+        System.out.println(String.format("%s%11s : %10d %6.2f%%", indent,
+            bigNumberForSize(Double.valueOf(Math.pow(10, i)).longValue()), countBuckets[i],
```
Comment (Manno15) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java:112]:
```suggestion
            NumUtil.bigNumberForSize(Double.valueOf(Math.pow(10, i)).longValue()), countBuckets[i],
```

**Comment by DomGarguilo** [comment]
It looks like with these changes the output is something like:
Screenshot from 2021-08-03 15-48-14

Is this what we are looking for?

**Comment by Shivakumar-SG** [comment]
> It looks like you are missing the import for `bigNumberForSize()` which is `import org.apache.accumulo.core.util.NumUtil`

Initially, I did import initially and when the command `mvn package -DskipTests` is run, it takes it out.

**Comment by Manno15** [comment]
> Initially, I did import and when the command mvn package -DskipTests is run, it takes it out.

That is most likely due to the import being unused since you weren't calling the function correctly (as suggested above). One of the format rules removes all unused imports.

**Comment by Manno15** [comment]
> Is this what we are looking for?

That output looks good to me.

File: core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java [hunk 2]
```diff
@@ -106,10 +107,11 @@ public void add(int size) {
...
 
     public void print(String indent) {
-      System.out.println(indent + "Up to size      count      %-age");
+      System.out.println(indent + "Up to size      Count      %-age");
       for (int i = 1; i < countBuckets.length; i++) {
-        System.out.println(String.format("%s%11.0f : %10d %6.2f%%", indent, Math.pow(10, i),
-            countBuckets[i], sizeBuckets[i] * 100. / totalSize));
+        System.out.println(String.format("%s%11s : %10d %6.2f%%", indent,
+            NumUtil.bigNumberForSize(Double.valueOf(Math.pow(10, i)).longValue()), countBuckets[i],
+            sizeBuckets[i] * 100. / totalSize));
```
Comment (Manno15) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java:114]:
```suggestion
        System.out.println("%s%11s : %10d %6.2f%%", indent,
            NumUtil.bigNumberForSize(Double.valueOf(Math.pow(10, i)).longValue()), countBuckets[i],
            sizeBuckets[i] * 100. / totalSize);
```
Based from my testing, the String.format isn't necessary.

**Review by Manno15 — CHANGES_REQUESTED** [review]
`.DS_Store` binary file should be removed from this pull request before this can be approved.

**Comment by EdColeman** [comment]
For discussion on output format - it looks like there are some options - using powers pr 10 or 2.  To produce a sample:

```
    for (int x = 1; x < 15; x++) {
      System.out.println(
          String.format("%s%11s", " ", NumUtil.bigNumberForQuantity((long) Math.pow(10, x))));
    } 
    System.out.println("---");
    for (int x = 1; x < 16; x++) {
      System.out.println(
          String.format("%s%11s", " ", NumUtil.bigNumberForSize((long) Math.pow(16, x))));
    }
```
This produces:
```
          10
         100
       1.00K
      10.00K
     100.00K
       1.00M
      10.00M
     100.00M
       1.00B
      10.00B
     100.00B
       1.00T
      10.00T
     100.00T
---
          16
         256
       4.00K
      64.00K
       1.00M
      16.00M
     256.00M
       4.00G
      64.00G
       1.00T
      16.00T
```
Having even numbers seems more pleasing (to me) base 2 or 10 work, but being the buckets are calculated using 100, bigNumberForQuantity might be easier to implement with this PR.  Being this is size, an argument for the second option is valid, but might require additional work to calculate the buckets correctly.

File: core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java [hunk 3]
```diff
@@ -106,10 +107,11 @@ public void add(int size) {
     }
 
     public void print(String indent) {
-      System.out.println(indent + "Up to size      count      %-age");
+      System.out.println(indent + "Up to size      Count      %-age");
       for (int i = 1; i < countBuckets.length; i++) {
-        System.out.println(String.format("%s%11.0f : %10d %6.2f%%", indent, Math.pow(10, i),
-            countBuckets[i], sizeBuckets[i] * 100. / totalSize));
+        System.out.println("%s%11s : %10d %6.2f%%", indent,
```
Comment (Manno15) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java]:
```suggestion
        System.out.printf("%s%11s : %10d %6.2f%%", indent,
```

That is my mistake, I forgot to add this change.

**Comment by Manno15** [comment]
> Having even numbers seems more pleasing (to me) base 2 or 10 work, but being the buckets are calculated using 100, bigNumberForQuantity might be easier to implement with this PR. Being this is size, an argument for the second option is valid, but might require additional work to calculate the buckets correctly.

Good point, I concur that Quantity might be preferable since it lines up closer to the original output.

**Comment by Manno15** [comment]
@Shivakumar-SG What are your thoughts on the conversation above? 

> Having even numbers seems more pleasing (to me) base 2 or 10 work, but being the buckets are calculated using 100, bigNumberForQuantity might be easier to implement with this PR. Being this is size, an argument for the second option is valid, but might require additional work to calculate the buckets correctly.

**Comment by Shivakumar-SG** [comment]
> @Shivakumar-SG What are your thoughts on the conversation above?
> 
> > Having even numbers seems more pleasing (to me) base 2 or 10 work, but being the buckets are calculated using 100, bigNumberForQuantity might be easier to implement with this PR. Being this is size, an argument for the second option is valid, but might require additional work to calculate the buckets correctly.

@Manno15 Thank you for showing your interest in my views but I'm just a student who is still learning and knows pretty much nothing about this repo. So I don't think so that I'm fit to share my views on such a big project.
I will just do what you tell me to do sir!

File: core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java [hunk 4]
```diff
@@ -106,10 +107,11 @@ public void add(int size) {
...
 
     public void print(String indent) {
-      System.out.println(indent + "Up to size      count      %-age");
+      System.out.println(indent + "Up to size      Count      %-age");
       for (int i = 1; i < countBuckets.length; i++) {
-        System.out.println(String.format("%s%11.0f : %10d %6.2f%%", indent, Math.pow(10, i),
-            countBuckets[i], sizeBuckets[i] * 100. / totalSize));
+        System.out.printf("%s%11s : %10d %6.2f%%", indent,
+            NumUtil.bigNumberForSize(Double.valueOf(Math.pow(10, i)).longValue()), countBuckets[i],
+            sizeBuckets[i] * 100. / totalSize);
```
Comment (Manno15) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java:114]:
```suggestion
        System.out.println(String.format("%s%11s : %10d %6.2f%%", indent,
            NumUtil.bigNumberForQuantity((long) Math.pow(10, i)), countBuckets[i],
            sizeBuckets[i] * 100. / totalSize));
```

**Review by Manno15 — APPROVED** [review]
Looks great.

**Review by DomGarguilo — APPROVED** [review]
LGTM. Output is as follows and looks good:
```
Up to size      Count      %-age
         10 :     500212   3.04%
        100 :   13459310  96.96%
      1.00K :          0   0.00%
     10.00K :          0   0.00%
    100.00K :          0   0.00%
      1.00M :          0   0.00%
     10.00M :          0   0.00%
    100.00M :          0   0.00%
      1.00B :          0   0.00%
     10.00B :          0   0.00%
```

**Review by EdColeman — APPROVED** [review]

**Comment by Manno15** [comment]
Thanks for the contribution @Shivakumar-SG and welcome to the Apache Accumulo community! If you would like to be listed as a contributor on Accumulo's people page please make a website PR for people.md. https://github.com/apache/accumulo-website/edit/main/pages/people.md

If you intend to make more contributions, please consider subscribing to the dev list and introducing yourself. Also, you can use the dev list or slack to reach out if you have questions about anything. https://accumulo.apache.org/contact-us/

**Comment by Shivakumar-SG** [comment]
I will consider your options. Thank you so much for letting me do this.

**Comment by EdColeman** [comment]
Thanks for the contribution Shivakumar-SG and welcome to the Apache Accumulo community. If you would like to be listed as a contributor on Accumulo's people page please make a website PR for people.md. https://github.com/apache/accumulo-website/edit/main/pages/people.md by:
 -  adding your name (in alphabetical order) to the contributor's list.
 -  reference the PR number (#2220)
If you intend to make more contributions, please consider subscribing to the dev list and introducing yourself. Also, you can use the dev list or slack to reach out if you have questions about anything. https://accumulo.apache.org/contact-us/

**Comment by Shivakumar-SG** [comment]
Thank you for helping me out in this @EdColeman and @DomGarguilo :)

**Pull request merged**